### PR TITLE
Creates CONTRIBUTOR and MAINTAINERS files

### LINKED ISSUE #349: Create CONTRIBUTING.md and MAINTAINERS.md
I think it would be a good idea to add the following files

CONTRIBUTING.md
==
This will be used to detail what future contributors can expect from the project.  

These are the normal items like commits must have DCO, all existing tests should pass, and new functionality should have tests, and a peer review by @ruby-git/maintainers

MAINTAINERS.md
==
Can be used by the community to know who needs to review the pull request that are created

### PR DESCRIPTION
**Pull request by tarcinil**
Created to create a standard review expectation for potential contributors and a clear representation of maintainers. 

The MAINTAINERS file will help avoid issues like those created in https://github.com/ruby-git/ruby-git/pull/342#issuecomment-368945651

closes #349

### CODE REVIEW

**Comment by tarcinil** [comment]
The failure of jruby-9.1.15.0 in Travis CI seems to be a random occurrence.  My change doesn't even change how the code works but it failed the test.  

I would propose that we allow failures of jruby-9.1.15.0 in Travis.  What do you think @perlun?

**Comment by perlun** [comment]
> The failure of jruby-9.1.15.0 in Travis CI seems to be a random occurrence. My change doesn't even change how the code works but it failed the test.

You're right, it seems to be an intermittent error. I restarted the job now.

> I would propose that we allow failures of jruby-9.1.15.0 in Travis. What do you think @perlun?

I am hesitant to do that, since it would easily allow us to miss real JRuby breakage (i.e. things that for whatever reason doesn't work there.) I'm a long-time JRuby user myself, and I know how annoying it is with gems that only work on MRI, so for the time being, I think it would be much better to look into the root cause for that intermittent error. There is an issue about it: https://github.com/ruby-git/ruby-git/issues/319

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  * [Pull Request Requirements](#pull-request-requirements)
+  * [Code Review Process](#code-review-process)
+  * [Developer Certification of Origin (DCO)](#developer-certification-of-origin-dco)
+
+
+## How Can I Contribute?
+
+### Submitting Issues
+
+We utilize **Github Issues** for issue tracking and contributions. You can contribute in two ways:
```
Comment (perlun) [inline CONTRIBUTING.md]: Minor nit: Github -> GitHub, both here and in the other places in the document.

**Comment by tarcinil** [comment]
I will rebase the branch when I get home.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    person who certified (a), (b) or (c) and I have not modified
+    it.
+
+(d) I understand and agree that this project and the contribution
+    are public and that a record of the contribution (including
+    all personal information I submit with it, including my
+    sign-off) is maintained indefinitely and may be redistributed
+    consistent with this project or the open source license(s)
+    involved.
+```
```
Comment (perlun) [inline CONTRIBUTING.md:94]: @rvodden I'm not so sure we need the DCO. It feels a bit far-fetched to me. What do you think?

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+### Your checklist for this pull request
+🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.
+
+- [ ] Ensure all commits include
```
Comment (perlun) [inline PULL_REQUEST_TEMPLATE.md]: include what?

Comment (tarcinil) [reply PULL_REQUEST_TEMPLATE.md]: It should have been the DCO sign-off, but could be rendered moot by your other comments.

File: CONTRIBUTING.md (hunk 1 above)
Comment (tarcinil) [reply CONTRIBUTING.md]: I will do tonight.

File: PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (perlun) [reply PULL_REQUEST_TEMPLATE.md]: @tarcinil Fair enough. Ping me or @rvodden once you have it rebased and we'll give it another look.

File: CONTRIBUTING.md (hunk 2 above)
Comment (rvodden) [reply CONTRIBUTING.md:94]: I'm with you, in that it is a bit far fetched, but on the other hand all the OS projects i work on them have them... its only a check box, and I don't think its much to ask; On the other hand if we miss it out and have no legal protection at all we risk (admittedly and extremely low risk) losing all our work. I'm minded to leave it in...

File: PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (tarcinil) [reply PULL_REQUEST_TEMPLATE.md]: @perlun My branch has been rebased and comments addressed.

The thread safety issue has also struck on the Ruby 2.1.10 inside of Travis.

File: CONTRIBUTING.md (hunk 2 above)
Comment (perlun) [reply CONTRIBUTING.md:94]: I agree about the legal implications, they are valid. _If_ it was only a check box, I would be fine with it, but as we have it written now:

> In the commit message of the contribution, the developer simply adds a Signed-off-by statement and thereby agrees to the DCO, which you can find below or at <http://developercertificate.org/>. 

...it means more work _every time someone makes a single commit_ and submits it to the project. (OTOH, you can automate it by using `git commit -s` or `--sign-off`, but still...)

I think, if we rephrase this to be just a mere checkbox in the PR, I am fine with it.

Comment (tarcinil) [reply CONTRIBUTING.md:94]: @perlun I would be hesitate to change the standard for DCO without understand the legal implications that change could introduce.  The DCO depends on the git Sign-Off-By because of the integrity that is built into Git, removing that could nullify the protections that are given to the maintainers of the project.

The IDE that I use has the ability to sign-off a commit built-in, but I do not believe that the addition of a `-s` in a command line string is asking too much either.  GitHub has tools that can check for DCO on a commit.

Comment (perlun) [reply CONTRIBUTING.md:94]: > The IDE that I use has the ability to sign-off a commit built-in, but I do not believe that the addition of a -s in a command line string is asking too much either. GitHub has tools that can check for DCO on a commit.

Alright, I'm willing to give it a try. We can drop this requirement later if it turns out to be too burdensome.

**Comment by perlun** [comment]
@tarcinil I merged this now. Please add the tool in GitHub that checks for DCO so we don't have to check that manually.

**Comment by tarcinil** [comment]
@perlun I would love to but I believe I would need to be added to the organization so that I can configure integrations. 

Please see the integration located here:
https://github.com/probot/dco

Installation can be done through:
https://github.com/integration/dco

**Comment by perlun** [comment]
@tarcinil Sure, I'll add you - we need all the help we can get. Visit this page and you should be able to see your invitation: https://github.com/orgs/ruby-git/invitation

**Comment by tarcinil** [comment]
@perlun A request has been sent to install it.

> A request to install DCO has been submitted on the @ruby-git account.

**Comment by perlun** [comment]
@tarcinil Great. I approved the request now, thanks for your efforts.
